### DISCUSSION TITLE
Disabling JIT in development in TailwindCSS v3?

### DISCUSSION THREAD

**Discussion by shawnwildermuth** [tailwindlabs/tailwindcss#7372] [Help] [answered]
First of all, I love JIT but with large projects it is taking a long time to parse through the used classes. I'd rather have an option to create a .css will *all* classes that I use when prototyping so when I add a new class, I don't have to wait 5+ seconds for the update to happen. I'm using --watch so maybe it's not working as expected. Here is my config:

```
module.exports = {
  content: ["./**/*.{html,njk}", "./client/**/*.vue"],
  safelist: [
    { pattern: /embed-/ },
    { pattern: /grid-/ },
    { pattern: /gap-/ },
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: ["Roboto", "Helvetica Neue", "Helvetica", "Arial", "sans-serif"],
        serif: ["Georgia", "Times New Roman", "serif"],
      },
      animation: {
        "spin-slow": "spin 2s linear infinite",
      },
    },
  },
  variants: {
    extend: {
      animation: ["hover"],
    },
  },
  plugins: [],
};
```

And I'm running this:

```
tailwind build -i ./src/wilderblog.src.css -o ./_site/css/wilderblog.min.css --watch
```

**Comment by alexanderbluhm** [comment]
Hi @shawnwildermuth 
As far as I know it is not possible to generate the full CSS in V3.0+.
One problem could be that you use `"./**/*.{html,njk}"` as a content path which basically searches for Tailwind classes in your entire directory. So, it also scans your `node_modules` folder which probably contains very many files.
Instead you should do something like `'./src/**/*.{html,njk}'` or at least exclude the `node_modules`.
You can read more about the content recommendations [here](https://tailwindcss.com/docs/content-configuration#pattern-recommendations)

**Reply by shawnwildermuth** [reply]
Might be the issue, let me check to see if it's searching that. Might explain the perf issue, but I do have a lot of files so having an option to build without JIT (or include all classes) for prototyping pages might really be useful.

**Reply by shawnwildermuth** [reply]
I'll close this, as the exclusion of node_modules really improved the perf enough to not need it. (5 seconds down to .5 seconds).

**Comment by chiptus** [comment]
my problem with not having all classes available is that I have to change the class name in my code, and not just play with different classes in the dev tools. is there something around this? maybe an extension?

**Comment by mpociot** [comment]
@chiptus That's exactly the reason why we built the Devtools for Tailwind chrome extension: https://devtoolsfortailwind.com/

It allows you to make use of all classes (and even the JIT engine) right in your devtools

**Reply by chiptus** [reply]
it's nice, but it's a paid solution and I need to leave the regular design tools.

**Comment by adamwathan** [comment]
Tailwind 3 has no way to generate all classes simply because the number of classes Tailwind 3 is capable of is far too high.

This is a valid class in Tailwind 3:

```
dark:lg:hover:focus:active:group-disabled:portrait:even:motion-reduce:bg-black
```

As an experiment we tried to see what would happen if we generated everything, and the node process crashed after the CSS file reached 4GB in size. There's simply no way the browser would load that file in development even if it worked.

If you want to generate a specific subset of classes, your best option is to use safelisting to list all of the class/patterns (along with the specific variants you want to generate) in your config file.

See previous discussion here: https://github.com/tailwindlabs/tailwindcss/discussions/6256#discussioncomment-1876701

**Comment by sntran** [comment, marked as answer]
I have some fun using Tailwind Play CDN for prototyping, as I can literally use any utilities available from my config file.

However, the lack of full plugin support and not being able to use external CSS file (#7147 ) make it a little hard to switch from development to production.

**Reply by shawnwildermuth** [reply]
I realized that this was by far the simplest solution for my needs

**Comment by await-ovo** [comment]
I wrote a plugin to solve the need for debugging styles in Chrome Devtools during development, and so far I feel very comfortable using this plugin with Tailwind CSS v3:

https://github.com/await-ovo/tailwindcss-webpack-plugin
